backend/src/test/java/com/rota/api/EngagementTests.java

```diff
@@ -0,0 +1,88 @@
+package com.rota.api;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+import com.rota.api.dto.EngagementDto;
+import com.rota.database.orm.engagement.Engagement;
+import com.rota.database.orm.engagement.EngagementRepository;
+import com.rota.database.orm.engagement.EngagementType;
+import com.rota.database.orm.staff.Staff;
+
+import java.time.Instant;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class EngagementTests {
```

Comment:
Good effort adding tests, I need to add some for methods I've created but I've been a bit lazy!

We could add tests for no engagements being returned and some negative tests such as invalid dates, start date after end date etc.